meta-ostro/classes/ostro-version.bbclass

```diff
@@ -0,0 +1,56 @@
+# When building in a CI system with swupd enabled, OS_VERSION must be
+# set to a consistent value for all builds. See below
+# for an example how that works with Jenkins.
+#
+# In local builds, we only have ${DATETIME} as something that
+# increments automatically, but it is too large for an integer number
+# on 32 bit systems. Therefore we substract the 2016 as the initial
+# year in which Ostro OS started using swupd and ignore the
+# seconds.
+#
+# The default behavior is to not rebuild just because OS_VERSION
+# changed. If that is desired, include in local.conf:
+#   VERSION_ID = "${OS_VERSION}"
+# This will cause the os-release package to be rebuilt each time
+# OS_VERSION changes, and that in turn causes an image and bundle
+# rebuild.
+#
+# For more predictable results is possible to set OS_RELEASE manually,
+# either in local.conf or in the environment like this:
+#   BB_ENV_EXTRAWHITE="$BB_ENV_EXTRAWHITE OS_VERSION" OS_VERSION=110 bitbake ...
+
+def ostro_get_os_version(d):
+    build_id_str = d.getVar('BUILD_ID', True) or ''
+    if build_id_str:
+        try:
+            build_id = int(build_id_str)
+        except ValueError:
+            # Assume that BUILD_ID is in CI's format, e.g. "2016-07-10_18-49-16-build-127".
+            # We cannot have the definition for OS_VERSION in ostroproject-ci.inc because
+            # the file is used also in eSDKs produced by CI where BUILD_ID is not set in
+            # the expected format.
```

Comment:
@rojkov can't we exclude auto.conf from the packaged eSDK (but beware of the MACHINE value, see below)? Please check with Paul Eggleton.

Related: we publish four eSDK archives, one per target platform. I suspect that this can be replaced with just a single one where developers then use settings in their own, custom local.conf or auto.conf to choose the actual machine they are interested in. Or do major parts of the cross-compile toolchain already get included in the ostro-x86_64-corei7-64-toolchain-ext-1.0~snapshot.sh archive?

I.e., how target architecture-specific is it really, besides including an auto.conf where the CI has changed the MACHINE value?